File: app.Tabaldi.PACT.Domain/ClientsModule/ClientAgg/Commands/ClientAddCommand.cs
using System;
using System.Collections.Generic;
using app.Tabaldi.PACT.Domain.AttendanceModule.AttendanceAgg;
using app.Tabaldi.PACT.Domain.AttendanceModule.AttendanceRecurrenceAgg.Commands;

namespace app.Tabaldi.PACT.Domain.ClientsModule.ClientAgg.Commands
{
    public class ClientAddCommand
    {
        public string Name { get; set; }
        public string Phone { get; set; }
        public DateTime DateOfBirth { get; set; }
        public decimal Value { get; set; }
        public ChargingType ChargingType { get; set; }
        public string ClinicalDiagnosis { get; set; }
        public string PhysiotherapeuticDiagnosis { get; set; }
        public string Objectives { get; set; }
        public string TreatmentConduct { get; set; }
        public IEnumerable<AttendanceRecurrenceCommand> Recurrences { get; set; }
    }
}

File: Mobile/FisioTech/app/src/main/java/com/tabaldi/fisiotech/ui/clients/ClientAddCommand.java
package com.tabaldi.fisiotech.ui.clients;

import com.tabaldi.fisiotech.base.Utils;

import java.util.ArrayList;
import java.util.List;

public class ClientAddCommand {
    public String name;
    public String phone;
    public String dateOfBirth;
    public String value;
    public ChargingType chargingType;
    public String clinicalDiagnosis;
    public String physiotherapeuticDiagnosis;
    public String objectives;
    public String treatmentConduct;
    public List<AttendanceAddRecurrenceCommand> recurrences;

    public void AddRecurrence(WeekDay weekDay, String startTime, String endTime) {
        AttendanceAddRecurrenceCommand recurrence = new AttendanceAddRecurrenceCommand();
        recurrence.weekDay = weekDay;
        recurrence.startTime = Utils.formatApiDate(Utils.stringToDate(startTime, Utils.VIEW_TIME_FORMAT));
        recurrence.endTime = Utils.formatApiDate(Utils.stringToDate(endTime, Utils.VIEW_TIME_FORMAT));

        if (this.recurrences == null) this.recurrences = new ArrayList<>();

        this.recurrences.add(recurrence);
    }
}
